code/modules/projectiles/guns/rifles.dm

```diff
@@ -815,3 +815,77 @@
 		to_chat(user, "<span class='warning'>There's no trigger on this gun, you have no idea how to fire it!</span>")
 		return FALSE
 	return TRUE
+
+
+//-------------------------------------------------------
+//TL-127 bolt action sniper rifle
+
+/obj/item/weapon/gun/rifle/chambered
+	name = "\improper TL-127 bolt action rifle"
+	desc = "The TL-127 is the standard issue bolt action rifle used by the TGMC. Known for its long range accuracy and use by marksmen despite its age and lack of IFF. It has an irremoveable scope. Uses 8.6×70mm box magazines."
+	icon = 'icons/Marine/gun64.dmi'
+	icon_state = "tl127"
+	item_state = "tl127" //thank you Alterist
+	fire_sound = 'sound/weapons/guns/fire/tl127.ogg'
+	dry_fire_sound = 'sound/weapons/guns/fire/sniper_empty.ogg'
+	unload_sound = 'sound/weapons/guns/interact/m41a_unload.ogg'
+	reload_sound = 'sound/weapons/guns/interact/m41a_reload.ogg'
+	caliber = "8.6×70mm" //codex
+	max_shells = 10 //codex
+	current_mag = /obj/item/ammo_magazine/rifle/chamberedrifle
+	type_of_casings = "cartridge"
+	attachable_allowed = list(
+						/obj/item/attachable/bayonetknife,
+						/obj/item/attachable/scope/unremovable/tl127,
+						/obj/item/attachable/extended_barrel,
+						/obj/item/attachable/heavy_barrel,
+						/obj/item/attachable/suppressor,
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/bayonetknife,
+						/obj/item/attachable/compensator)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_WIELDED_FIRING_ONLY|GUN_CAN_POINTBLANK
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO)
+	attachable_offset = list("muzzle_x" = 41, "muzzle_y" = 19,"rail_x" = 8, "rail_y" = 21, "under_x" = 37, "under_y" = 16, "stock_x" = 9, "stock_y" = 12)
+	starting_attachment_types = list(/obj/item/attachable/scope/unremovable/tl127,
+									/obj/item/attachable/stock/tl127stock)
+
+	fire_delay = 1.2 SECONDS
+	accuracy_mult = 1.35
+	accuracy_mult_unwielded = 0.7
+	scatter = -30
+	scatter_unwielded = 40
+	recoil = 0
+	recoil_unwielded = 4
+	aim_slowdown = 1
+	wield_delay = 1 SECONDS
+
+	var/rack_delay = 7
+	var/rack_sound = "tl-127_bolt.ogg"
+	var/racked_bolt = TRUE
+	var/cooldown_time
+
+/obj/item/weapon/gun/rifle/chambered/able_to_fire(mob/user)
+	. = ..()
+	if(!racked_bolt)
+		to_chat(user, "<span class='notice'>[src] does not have a round chambered!</span>")
+		return FALSE
+	return TRUE
+
+/obj/item/weapon/gun/rifle/chambered/unique_action(mob/user)
+	if(racked_bolt)
+		to_chat(user, "<span class='notice'>[src] already has a round chambered!</span>")
+		return
+	if(!(cooldown_time < world.time))
+		return
```

Comment:
Replace this with `COOLDOWN_CHECK`

`COOLDOWN_CHECK(src, COOLDOWN_GUN_CHAMBER)` would work if you add the define